File: authorization-manager-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__OAuthAuthorizationManager.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.attribute.ContextAttributes;
import se.curity.identityserver.sdk.attribute.SubjectAttributes;
import se.curity.identityserver.sdk.authorization.AuthorizationResult;
import se.curity.identityserver.sdk.authorization.OAuthObligation;
import se.curity.identityserver.sdk.authorization.oauth.OAuthAuthorizationActionAttributes;
import se.curity.identityserver.sdk.authorization.oauth.OAuthAuthorizationManager;
import se.curity.identityserver.sdk.authorization.oauth.OAuthAuthorizationResourceAttributes;

public final class ${pluginName}OAuthAuthorizationManager implements OAuthAuthorizationManager
{
    private final ${pluginName}AuthorizationManagerConfig _configuration;

    public ${pluginName}OAuthAuthorizationManager(${pluginName}AuthorizationManagerConfig configuration)
    {
        _configuration = configuration;
    }

    @Override
    public AuthorizationResult<OAuthObligation> getOAuthAuthorizationResult(SubjectAttributes subjectAttributes,
                                                                            OAuthAuthorizationActionAttributes oAuthAuthorizationActionAttributes,
                                                                            OAuthAuthorizationResourceAttributes oAuthAuthorizationResourceAttributes,
                                                                            ContextAttributes contextAttributes)
    {
        // TODO: Perform decision and return appropriate AuthorizationResult
        return AuthorizationResult.notApplicable();
    }

}

File: kotlin-authorization-manager-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__OAuthAuthorizationManager.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.attribute.ContextAttributes
import se.curity.identityserver.sdk.attribute.SubjectAttributes
import se.curity.identityserver.sdk.authorization.OAuthObligation
import se.curity.identityserver.sdk.authorization.oauth.OAuthAuthorizationActionAttributes
import se.curity.identityserver.sdk.authorization.oauth.OAuthAuthorizationManager
import se.curity.identityserver.sdk.authorization.oauth.OAuthAuthorizationResourceAttributes

class ${pluginName}OAuthAuthorizationManager(private val _configuration: ${pluginName}AuthorizationManagerConfig): OAuthAuthorizationManager
{
    override fun  getOAuthAuthorizationResult(subjectAttributes: SubjectAttributes ,
                                     oAuthAuthorizationActionAttributes: OAuthAuthorizationActionAttributes ,
                                     oAuthAuthorizationResourceAttributes: OAuthAuthorizationResourceAttributes ,
                                     contextAttributes: ContextAttributes)
        : AuthorizationResult<OAuthObligation>
    {
        // TODO: Perform decision and return appropriate AuthorizationResult
        TODO("Not implemented")
    }

}
